skytemple_files/_resources/patches/asm_patches/cipnit_asm_mods/completeteamcontrol.asm

```diff
@@ -0,0 +1,299 @@
+	bl @Mode_Switch	;original: ldrh r0,[r0,2h]
+
+.org NA_022EC350	;end of leader's turn verification algorithm
+	bl @WaitChecker	;original: ldrsh r0,[r0,r1]
+
+.org NA_022EC488	;Jump to the function which executes the leader's action
+	bl @Leader_Prepper	;original: bl NA_022FE4BC
+
+.org NA_022EBD78	;When the turns of your partners start
+	bl @Change_Leader	;original: mov r5,0h
+
+.org NA_022EC4C0	;Leader's wait bit setter - this is actually a bug in the original game, where the leader's wait bit would also raise when you trade places with a partner, which the developers never noticed because the game never checks the leader's wait bit. The wait bit is used to skip a partner's next turn when you trade places with it, which is a big deal in this hack since leadership is constantly changing.
+	nop		;original: orrne r0,r0,4000h
+
+.org NA_022F9920	;The pause between turns - changed so it's not ridiculously long
+	cmp r0,0Ch	;original: cmp r0,3Ch
+
+.org NA_022EC5F0	;Round counter resetter - nopping this prevents a bug when you get a speed boost, explained in a reply to this mod's pokecommunity post.
+	nop		;original: strneh r1,[r0,80h]
+
+.org NA_022F1178	;Sets the partner to look at you. Had to remove because it sets the pokemon's graphical direction but not its mechanical direction, so when you started manual mode, your partner would suddenly change directions.
+	nop		;original: bl NA_023055B0
+;if I ever want to fix this in the future:
+;direction you're facing is pokemon's nonentity data + 4C
+;0230567C: r0 is pokemon's entity address, r2 is new direction
+;I'm not fixing it because I think pokemon looking in different directions looks cool, but if I wanted to fix it so pokemon look at you at the start of every turn like in the original game, I've got to put a blop at 0230567C to custom code which corrects the problem
+
+.org NA_0230506C	;Another partner direction setter
+	b NA_02305160	;original: ldrsh r1,[r0,7Eh]
+;023050E4: r1 is new direction, r0 is pokemon's address
+
+.org NA_022F488C	;Equip penalty - when you give a pokemon an item in the original game, its turn is skipped if it's a partner, but if the leader gives itself an item, its next turn isn't skipped. This causes a bug where, if a partner gives itself an item, its next turn is skipped, and if a partner gives the leader an item, the leader's next turn isn't skipped. Fixed so it works fairly: a pokemon doesn't lose its next turn if it gives itself an item, but if it gives an item to another pokemon, the other pokemon's turn is skipped.
+	bl @EquipPenalty	;original: ldrb r0,[r9,7h]
+
+.org NA_022F499C	;All of the above also applies to taking items from pokemon
+	bl @DequipPenalty	;original: ldrb r0,[r5,7h]
+
+.close
+
+.open "overlay_0031.bin", ov_31
+.org NA_02387530	;Jump to team submenu function
+	bl @paranoia_agent	;original: bl NA_022EB408
+.close
+
+
+.open "overlay_0036.bin", ov_36
+.org CompleteTeamControl_RAMPosition

```

Comment:
```suggestion
.orga 0x10
.area 3F0h - 10h
```
No need to know the start offset of the extra overlay.
I'll leave 16 reserved bytes at the start of the file so we can add a header if it's ever needed, you can have 10h - 3F0h.
If you want some extra bytes just in case you need to make changes later let me know. The allocated space for your hack can be changed until somebody else requests another area inside the overlay.